File: DMOJ/CCC/15/ZeroThatOut.py
import sys


def main():
    s = []
    K = int(sys.stdin.readline())
    total = 0
    for i in range(K):
        temp = int(sys.stdin.readline())
        if (temp == 0):
            total -= s[-1]
            s.pop()
        else:
            s.append(temp)
            total += temp
    print(total)


main()

File: DMOJ/CCC/15/ZeroThatOut.java
import java.io.*;
import java.util.*;
public class ZeroThatOut {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int K = Integer.parseInt(br.readLine()), total = 0;
		Stack<Integer> s = new Stack<>();
		for (int i = 0; i < K; i++) {
			int temp = Integer.parseInt(br.readLine());
			if (temp == 0) {
				total -= s.peek();
				s.pop();
			} else {
				total += temp;
				s.push(temp);
			}
		}
		System.out.println(total);
	}
}

